hedera-node/src/main/java/com/hedera/services/state/submerkle/ExpirableTxnRecord.java

```diff
@@ -408,23 +408,30 @@ public void deserialize(SerializableDataInputStream in, int version) throws IOEx
 		}
 	}
 
-	private void deserializeAllowanceMaps(SerializableDataInputStream in) throws IOException {
-		var numCryptoAllowances = in.readInt();
-		if (numCryptoAllowances > 0) {
-			cryptoAllowances = new TreeMap<>();
-		}
-		while (numCryptoAllowances-- > 0) {
-			final EntityNum owner = EntityNum.fromLong(in.readLong());
-			cryptoAllowances.put(owner, deserializeCryptoAllowances(in));
-		}
+	private void deserializeAllowanceMaps(SerializableDataInputStream in, final int version) throws IOException {
+		if (version < RELEASE_0250_VERSION) {
+			// In release 0.24.x three _always-empty_ map sizes were serialized here
+			in.readInt();
+			in.readInt();
```

Comment:
I think this is wrong. The original version had a readInt at the end of the while loop:
```
private void deserializeAllowanceMaps(SerializableDataInputStream in) throws IOException {
		var numCryptoAllowances = in.readInt();
		if (numCryptoAllowances > 0) {
			cryptoAllowances = new TreeMap<>();
		}
		while (numCryptoAllowances-- > 0) {
			final EntityNum owner = EntityNum.fromLong(in.readLong());
			cryptoAllowances.put(owner, deserializeCryptoAllowances(in));
		}

		var numTokenAllowances = in.readInt();
		if (numTokenAllowances > 0) {
			fungibleTokenAllowances = new TreeMap<>();
		}
		while (numTokenAllowances-- > 0) {
			final EntityNum owner = EntityNum.fromLong(in.readLong());
			fungibleTokenAllowances.put(owner, deserializeFungibleTokenAllowances(in));
		}
		var numNftAllowances = in.readInt();   // <---- this one.
	}
```

Reply:
Oops . And I realized I missed this extra while loop at the end if numNftAllowances is > 0:

```
		while (numNftAllowances-- > 0) {
			final EntityNum owner = EntityNum.fromLong(in.readLong());
			nftAllowances.put(owner, deserializeNftAllowances(in));
		}
```